src/test/CMakeLists.txt

```diff
@@ -134,7 +134,7 @@ endif(WITH_RADOSGW)
 if(WITH_LIBCEPHFS)
   # From src/test/Makefile-client.am: I dont get this one... testing the osdc build but link in libcephfs?
   add_executable(test_build_libcephfs buildtest_skeleton.cc)
-  target_link_libraries(test_build_libcephfs cephfs pthread ${CRYPTO_LIBS} ${EXTRALIBS})
+  target_link_libraries(test_build_libcephfs cephfs pthread stdc++ ${CRYPTO_LIBS} ${EXTRALIBS})
```

Comment:
if https://shaman.ceph.com/builds/ceph/wip-crimson-only-gcc11/10162591cb6969482d35a002dc6d59192484e7b4/crimson/308608/ compiles, does this implies that we don't need this change anymore?

also, i don't really understand why we need this change. right, libstdc++ includes the missing symbol. but why do we need to link against it explicitly? AFAIU, CMake should set `LINKER_LANGUAGE` for `test_build_libcephfs`, so presumably, it should link against the standard library, right?

also, this does not work if the user wants to use libc++.

Reply:
This issue pops only in some runs and looks like being machine-dependent. It always replicates on my `o06` env. It also replicated in CI/CD:

```
[ 70%] Linking CXX executable ../../bin/test_build_libcephfs
/opt/rh/gcc-toolset-11/root/usr/bin/ld: /opt/rh/gcc-toolset-11/root/usr/lib/gcc/x86_64-redhat-linux/11/libstdc++_nonshared.a(cow-stdexcept80.o): undefined reference to symbol '_ZTVSt11logic_error@@GLIBCXX_3.4'
```

https://jenkins.ceph.com/job/ceph-dev-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/49886//consoleFull

Reply:
Me neither. A linker's bug?

BTW: just got an info from @ronen-fr it happens also on GCC-10.

Reply:
@ronen-fr and @rzarzynski , could you please take a look at #47029? i think it is a better fix.